go/src/koding/kites/kloud/stack/auth.go

```diff
@@ -23,23 +28,57 @@ type LoginResponse struct {
 	GroupName string `json:"groupName"`
 }
 
+// Pinger send ping requests for presence.
+type Pinger interface {
+	Ping(string, string) error
+}
+
 // AuthLogin creates a jSession for the given username and team.
 //
 // If a session already exists, the method is a nop and returns
 // already existing one.
 //
 // TODO(rjeczalik): Add AuthLogout to force creation of a new
 // session.
-func (k *Kloud) AuthLogin(r *kite.Request) (interface{}, error) {
-	// TODO:
-	//
-	// - use "kd-io" if req.GroupName is empty
-	// - check if req.GroupName exists
-	// - check whether user belongs to req.GroupName
-	// - check whether group's subscription is active
-	// - check whether there exists a jSession for {GroupName, Username}
-	//   and return clienId if it does
-	// - create new jSession and return it's clientId
-	//
-	return nil, nil
+func (k *Kloud) AuthLogin(p Pinger) func(r *kite.Request) (interface{}, error) {
+	return func(r *kite.Request) (interface{}, error) {
+		k.Log.Debug("auth login called by %q with %q", r.Username, r.Args.Raw)
+
+		req, err := getLoginReq(r)
+		if err != nil {
+			return nil, err
+		}
+
+		ses, err := modelhelper.UserLogin(r.Username, req.GroupName)
+		if err != nil {
+			return nil, NewError(ErrInternalServer)
```

Comment:
Can we also debug log any error that was shadowed? (the ones hidden from the user)